lib/animas/animasSimulator.js

```diff
@@ -119,13 +168,30 @@ exports.make = function(config){
       }
 
       if(currStatus !== null) {
-        if(currStatus.status === 'suspended' && event.rate === 0) {
-          //this basal is suspended
-          event.deliveryType = 'suspend';
+        if(currStatus.status === 'suspended') {
+          if (event.rate === 0) {
+            //this basal is suspended
+            event.deliveryType = 'suspend';
+          }
+          else {
+            // basal is non-zero, but status still suspended
+            // is this a suspend event we created from an alarm?
+            // If so, we should submit a joined suspend/resume event
```

Comment:
Yes, the rest of the joined suspend/resume event functionality is in #257 which I just opened.